core-tests/src/test/java/org/mule/runtime/core/message/DefaultMuleMessageBuilderTestCase.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.message;
+
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mule.runtime.api.metadata.DataType.BOOLEAN;
+import static org.mule.runtime.api.metadata.DataType.HTML_STRING;
+import static org.mule.runtime.api.metadata.DataType.STRING;
+import static org.mule.runtime.api.metadata.DataType.TEXT_STRING;
+import static org.mule.runtime.api.metadata.MediaType.TEXT;
+import static org.mule.runtime.api.metadata.MediaType.XML;
+import org.mule.runtime.core.DefaultMuleMessage;
+import org.mule.runtime.core.DefaultMuleMessageBuilder;
+import org.mule.runtime.core.api.MuleMessage;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.mule.tck.testmodels.fruit.Apple;
+import org.mule.tck.testmodels.fruit.Banana;
+import org.mule.tck.testmodels.fruit.Orange;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import javax.activation.DataHandler;
+
+import org.junit.Test;
+
+/**
+ *
+ */
+public class DefaultMuleMessageBuilderTestCase extends AbstractMuleTestCase
+{
+
+    private static final Apple TEST_ATTR = new Apple();
+    private static final Banana TEST_ATTR_2 = new Banana();
+    private static final String PROPERTY_KEY = "propertyKey";
+    private static final Serializable PROPERTY_VALUE = "propertyValue";
+    private static final Object REPLY_TO = new Orange();
+
+    @Test
+    public void createNewAPIMessageViaMessageInterface()
+    {
+        org.mule.runtime.api.message.MuleMessage<String, Apple> message;
+        message = org.mule.runtime.api.message.MuleMessage.builder().payload(TEST_PAYLOAD).attributes(TEST_ATTR)
+                .build();
+        assertThat(message, instanceOf(DefaultMuleMessage.class));
```

Comment:
is this instanceof assertion really necessary? It's not like the builder may return different types.